apps/aens/src/aens_state_tree.erl

```diff
@@ -171,6 +176,32 @@ name_list(#ns_tree{mtree = Tree}) ->
 -endif.
 
 
+-spec to_binary_without_backend(tree()) -> binary().
+to_binary_without_backend(#ns_tree{mtree = MTree, cache = Cache}) ->
+    MTBin = aeu_mtrees:serialize(MTree),
+    CacheBin = aeu_mtrees:serialize(Cache),
+    aec_object_serialization:serialize(
+        nameservice_mtree,
+        ?VSN,
+        serialization_template(?VSN),
+        [ {mtree, MTBin}
+        , {cache, CacheBin}]).
+
+-spec from_binary_without_backend(binary()) -> tree().
+from_binary_without_backend(Bin) ->
+    [ {mtree, MTBin}
```

Comment:
The cache should not be in the serialization since it is not under consensus (not reflected in the root hash). The cache is nowhere defined, and it is just an implementation detail to handle future events.